library/xml-model/src/main/java/org/cru/godtools/xml/model/Manifest.kt

```diff
@@ -137,31 +139,44 @@ class Manifest : BaseModel, Styles {
 
         // process any child elements
         var title: Text? = null
-        var categoriesData: List<Category>? = null
-        var pagesData: PagesData? = null
-        var resourcesData: Map<String?, Resource>? = null
-        var tipsData: List<Tip>? = null
-        while (parser.next() != XmlPullParser.END_TAG) {
-            if (parser.eventType != XmlPullParser.START_TAG) continue
-
-            when (parser.namespace) {
-                XMLNS_MANIFEST -> when (parser.name) {
-                    XML_TITLE -> title = Text.fromNestedXml(this, parser, XMLNS_MANIFEST, XML_TITLE)
-                    XML_CATEGORIES -> categoriesData = parseCategories(parser)
-                    XML_PAGES -> pagesData = parsePages(parser, parseFile)
-                    XML_RESOURCES -> resourcesData = parseResources(parser)
-                    XML_TIPS -> tipsData = parser.parseTips(parseFile)
+        val aemImports = mutableListOf<Uri>()
+        val categories = mutableListOf<Category>()
+        val resources = mutableListOf<Resource>()
+        val tips: List<Tip>
+        val tractPages: List<TractPage>
+        runBlocking {
+            val tipsTasks = mutableListOf<Deferred<Tip>>()
```

Comment:
So setting this to ```Deferred``` means, regardless of whether it fails or not we can retrieve the result using the ```awaitAll``` we are running below?

Reply:
`Deferred` means that the value may not be finished calculating yet, `.awaitAll()` will suspend the coroutine until `Deferred` has finished.